### PR TITLE
feat: execute_starknet_call

### PR DESCRIPTION
**Pull request by obatirou**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #853 

## What is the new behavior?

- add execute_starknet_call

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot-ssj/871)

### CODE REVIEW

File: crates/contracts/src/account_contract.cairo [hunk 1]
```diff
@@ -271,6 +272,23 @@ pub mod AccountContract {
...
+            self.ownable.assert_only_owner();
+            let kakarot_address = self.ownable.owner();
+            if call.to == kakarot_address && call.selector != selector!("get_starknet_address") {
+                return (false, KAKAROT_REENTRANCY.span());
+            }
+            let response = call_contract_syscall(call.to, call.selector, call.calldata);
+            if response.is_ok() {
+                return (true, response.unwrap().into());
+            }
+            return (true, response.unwrap_err().into());
```
Comment (enitrat) [inline crates/contracts/src/account_contract.cairo:290]: should success be false here?

```suggestion
            return (false, response.unwrap_err().into());
```

File: crates/contracts/src/account_contract.cairo [hunk 2]
```diff
@@ -271,6 +272,23 @@ pub mod AccountContract {
             self.ownable.assert_only_owner();
             self.Account_nonce.write(nonce);
         }
+
+        // @notice Used to preserve caller in Cairo Precompiles
+        // @dev Reentrency check is done for Kakarot contract, only get_starknet_address is allowed
+        //      for Solidity contracts to be able to get the corresponding Starknet address in their
+        //      calldata.
```
Comment (enitrat) [inline crates/contracts/src/account_contract.cairo:279]: we prefer using rustdoc format, so no `@notice` nor `@dev` tags, and three slashes

```
/// Used to preserve caller in Cairo Precompiles.
///
```

File: crates/contracts/tests/test_contract_account.cairo [hunk 3]
```diff
@@ -63,3 +70,62 @@ fn test_ca_storage() {
 
     assert(storage == expected_storage, 'wrong contract storage');
 }
+
+#[test]
```
Comment (enitrat) [inline crates/contracts/tests/test_contract_account.cairo:74]: i am going to wait a bit before merging this as i am currently migrating to sn-foundry and this will change

File: crates/contracts/src/account_contract.cairo (hunk 1 above)
Comment (obatirou) [reply crates/contracts/src/account_contract.cairo:290]: Feature parity with kkrt0 is no check on the return of the call https://github.com/kkrt-labs/kakarot/blob/ebcdf0b0167cb5b5d1cb2934761809cfc1d63902/src/kakarot/accounts/account_contract.cairo#L339 and return what is sent back

Comment (enitrat) [reply crates/contracts/src/account_contract.cairo:290]: In kkrt0 we cannot catch the result of the syscall, which is why we cannot return FALSE.
In practice, if the syscall fails, the transaction will revert. But later in the future, we will be able to catch them, and I think it is better to have this entrypoint return false in this case.

tldr; the err case will never be triggered until call errors can be caught, as it will make the tx fail

**Review by enitrat — APPROVED** [review]
